### DISCUSSION TITLE
Error while deploying smart contract to sepolia using foundry in the alchemy node

### DISCUSSION THREAD

**Discussion by YatharthPnwr** [Cyfrin/foundry-full-course-cu#1593] [Q&A] [answered]
When i try to deploy my  smart contract using alchemy on the sepolia mainnet, it gives out the following error

Error: 
**Failed to send transaction**
Context:
- server returned an error response: error code -32000: max priority fee per gas higher than max fee per gas

the command i used was 
`forge script script/DeploySimpleStorage.s.sol --rpc-url $SEPOLIA_RPC_URL --private-key $PRIVATE_KEY_ALCHEMY --broadcast`

I have correctly mentioned the SEPOLIA_RPC_URL and the private key for the txn.
dont know how to solve this . Help would be really appreciated !

**Comment by EngrPips** [comment]
Hello @yatharth-singh-panwar , is the `$PRIVATE_KEY_ALCHEMY` anyway related to alchemy ?

**Reply by YatharthPnwr** [reply]
No, the `$PRIVATE_KEY_ALCHEMY` is not related to alchemy. It's simply my private key on MetaMask. Although in hindsight, I realize that I should have named the variable something more appropriate, like `$PRIVATE_KEY`.

**Reply by EngrPips** [reply]
Oh i see, was wondering because of the variable name lol, do you have enough sepolia eth to cover gas cost on the wallet associated with the private key you provided ?

**Reply by YatharthPnwr** [reply]
yes i have plenty of sepolia eth in the account. In fact, i even got more from the faucet to ensure this and now i have more than I think i need

**Reply by EngrPips** [reply, marked as answer]
Okay, use the command below to deploy the smart contract, put the arguments required as needed 

```javascript
forge script/<name of your deployment scrip> --rpc-url <http://your rpc-url address> broadcast --private-key <your private-key>
```

Or you can just copy and run the command below 

```javascript
forge script script/DeploySimpleStorage.s.sol --rpc-url $SEPOLIA_RPC_URL broadcast --private-key $PRIVATE_KEY_ALCHEMY --
```
Basically the broadcast is suppose to come before the `--private-key` flag

**Reply by YatharthPnwr** [reply]
Hey ! @EngrPips . So today when i tried to execute the same command that i mentioned in the question , it worked without any issue. I tried both the ways, 1) writing --broadcast after private_key and 2) writing it before private_key, and both of them work as expected and my transaction went through. Don't know the exact reason though but thanks for helping out!

**Reply by EngrPips** [reply]
Sometime some things happen and we can't just explain them lol, i am glad you get it to work now. keep crushing them !

**Comment by LanceAddison** [comment]
If anyone else is still getting this error I found that this command worked for me. It's just adding --priority-gas-price 1 to the end of the command. 
```
forge script script/DeploySimpleStorage.s.sol --rpc-url $SEPOLIA_RPC_URL --private-key $PRIVATE_KEY --broadcast --priority-gas-price 1
```

**Reply by EngrPips** [reply]
Awesome, thanks for sharing this
